training/src/pages/LogIn/LogIn.jsx

```diff
@@ -0,0 +1,253 @@
+        });
+      }
+    });
+  };
+
+  forErrors = () => {
+    const { hasError, isTouched } = this.state;
+    const obj = Object.values(hasError);
+    const objForTouched = Object.values(isTouched);
+    let i;
+    for (i = 0; i < obj.length; i += 1) {
+      if (obj[i]) return false;
+    }
+    for (i = 0; i < objForTouched.length; i += 1) {
+      if (!objForTouched[i]) return false;
+    }
+    return true;
+  };
+
+  render() {
+    const { classes } = this.props;
+    const { errorOccured } = this.state;
+    console.log('State', this.state);
+    return (
+      <main className={classes.main}>
+        <CssBaseline />
+        <Paper className={classes.paper}>
+          <Avatar className={classes.avatar}>
+            <LockOutlinedIcon />
+          </Avatar>
+          <Typography component="h1" variant="h5">
+         Log In
+          </Typography>
+          <div>
+            <Grid container spacing={24} className={classes.field}>
+              <Grid item xs={12}>
+                <TextField
+                  id="outlined-password-input"
+                  label="Email"
+                  style={{ margin: '3px', width: 350 }}
+                  type="email"
+                  autoComplete="current-password"
+                  margin="normal"
+                  variant="outlined"
+                  onChange={this.selectHandler('email')}
+                  onBlur={this.selectEventHandler('email')}
+                  error={errorOccured.email}
+                  helperText={errorOccured.email}
+                  InputProps={{
+
+                    startAdornment: (
+                      <InputAdornment position="start">
+                        <IconButton>{<Email />}</IconButton>
+                      </InputAdornment>
+                    ),
+                  }}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  id="outlined-search"
+                  label="Password"
+                  type="password"
+                  margin="normal"
+                  variant="outlined"
+                  style={{ margin: '3px', width: 350 }}
+                  onChange={this.selectHandler('password')}
+                  onBlur={this.selectEventHandler('password')}
+                  error={errorOccured.password}
+                  helperText={errorOccured.password || ''}
+                  InputProps={{
+                    startAdornment: (
+                      <InputAdornment position="start">
+                        <IconButton
+                          aria-label="Toggle password visibility"
+                        >
+                          <VisibilityOff />
+                        </IconButton>
+                      </InputAdornment>
+                    ),
+                  }}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                { this.forErrors() ? (
+                  <Button
```

Comment:
You can apply the condition on disabled value rather duplicating whole Button component.